hw/ip/rom_ctrl/util/mem.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+import re
+import subprocess
+import tempfile
+from typing import BinaryIO, IO, List, Optional, TextIO, Tuple
+
+from elftools.elf.elffile import ELFFile  # type: ignore
+
+
+def red_xor32(word: int) -> int:
+    '''Reduction XOR for a uint32'''
+    word = (word & 0xffff) ^ (word >> 16)
+    word = (word & 0xff) ^ (word >> 8)
+    word = (word & 0xf) ^ (word >> 4)
+    word = (word & 0x3) ^ (word >> 2)
+    return (word & 0x1) ^ (word >> 1)
+
+
+def add_ecc32(word: int) -> int:
```

Comment:
do you want to leave a comment here that this can change if someone re-runs `secdec_gen` with a different seed?
The hsiao codes are sort of random... of course, you could also use hamming code then it would never change :)

Reply:
Oof, I didn't realise that. I'll re-write this to grab the random seed from the design files (eugh) and then call out to the internals of `secded_gen.py` to get the constants. It's a bit sad to need the design files, but I can't really see a better way. Anyway, we can always handle seed extraction better in future if we think of one.

Reply:
Honestly, baking the seed in is fine in this case I think.
If you want a completely deterministic code, you could also switch to standard Hamming encoding, which does not have a random element (it is a bit less optimal in terms of XOR gates, though).

Reply:
Cool, I'll add code to #6180 to get the seeds right here (since it will change them!)